src/components/drawer/drawer.pw.tsx

```diff
@@ -0,0 +1,488 @@
+    await footer.scrollIntoViewIfNeeded();
+    expect(await stickyFooter(page).count()).toBeGreaterThan(0);
+    await expect(stickyFooter(page)).toBeVisible();
+  });
+
+  test("should render component with custom stickyFooter", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <div
+        style={{
+          height: "200px",
+        }}
+      >
+        <DrawerCustomFooterHeader stickyFooter />
+      </div>
+    );
+
+    await expect(stickyFooter(page)).toBeVisible();
+  });
+
+  test("should render component with custom stickyHeader", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <div
+        style={{
+          height: "400px",
+        }}
+      >
+        <DrawerCustomFooterHeader stickyHeader />
+      </div>
+    );
+
+    const footer = stickyFooter(page);
+    await footer.scrollIntoViewIfNeeded();
+    await expect(
+      drawer(page).locator("div").locator("div").locator("div").nth(0)
+    ).toHaveText("Drawer title");
+    await expect(
+      drawer(page).locator("div").locator("div").locator("div").nth(0)
+    ).toBeVisible();
+  });
+});
+
+test.describe("check events for Drawer component", () => {
+  test("should call onChange callback when a component is closed", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <DrawerCustom
+        showControls
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const toggle = drawerToggle(page);
+    await toggle.click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onChange callback when a component is expanded", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <DrawerCustom
+        showControls
+        expanded={false}
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const toggle = drawerToggle(page);
+    await toggle.click();
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("Accessibility tests for Drawer", () => {
+  ([
+    ["expanded", 0],
+    ["not expanded", 1],
+  ] as [string, number][]).forEach(([state, times]) => {
+    test(`should pass accessibility tests when chevron is ${state}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<DrawerCustom showControls />);
+
+      for (let i = 0; i < times; i++) {
+        await drawerToggle(page).click();
+        await expect(drawerSidebar(page)).toBeVisible();
+      }
+
+      await checkAccessibility(page);
+    });
+  });
+
+  ["3s", "15s"].forEach((animationDuration) => {
+    test(`should pass accessibility tests when animation time is set to ${animationDuration}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <DrawerCustom showControls animationDuration={animationDuration} />
+      );
+
+      await drawerToggle(page).click();
+      await expect(drawerSidebar(page)).toBeVisible();
```

Comment:
**question** (non-blocking): same question here about whether the visibility assertion is actually needed or relevant to this particular test.